src/Engines/MeilisearchEngine.php

```diff
@@ -378,7 +373,28 @@ public function deleteIndex($name)
      */
     public function deleteAllIndexes()
     {
-        return $this->meilisearch->deleteAllIndexes();
+        $tasks = [];
+        $offset = 0;
+        $limit = 20;
+        $hasMorePages = true;
+
+        while ($hasMorePages) {
+            $query = new IndexesQuery();
+            $query->setOffset($offset)->setLimit($limit);
+
+            $indexes = $this->meilisearch->getIndexes($query);
+            foreach ($indexes->getResults() as $index) {
+                $tasks[] = $index->delete();
+            }
+
+            if ($indexes->count() < $limit) {
+                $hasMorePages = false;
+            }
+
+            $offset += $limit;
```

Comment:
⚠️ A big warning here

To explain the issue let's suppose $limit = 3; and you have 8 indexes in meilisearch named from A to H

A B C D E F G H

- A first request, you will delete A B C
- But here you changed the $offset to 3
- now meilisearch have index D E F G H

At the second request, you will actually delete G and H,
Now, because $indexes->count() (2) < 3, hasMorePages will becomes false,
indexes D E and F will remain on meilisearch

Reply:
Thats true, thanks for pointing this out 👍